test/unit/proxy_config_affecting_changes_test.rb

```diff
@@ -60,6 +60,38 @@ def self.column_names
     end
   end
 
+  class StoredModelTest < ActiveSupport::TestCase
+    class StoredModel < ActiveRecord::Base
+      include ProxyConfigAffectingChanges::ModelExtension
+      define_proxy_config_affecting_attributes :settings
+      self.table_name = 'gateway_configurations'
+
+      store :settings, accessors: %i[jwt_claim_with_client_id jwt_claim_with_client_id_type], coder: JSON
+    end
+
+    test 'tracks proxy config affecting changes on attribute write' do
+      within_thread do
+        tracker = ProxyConfigAffectingChanges::Tracker.new
+        Thread.current[ProxyConfigAffectingChanges::TRACKER_NAME] = tracker
```

Comment:
I see some usage for this in other tests as well (and indeed it is already being used elsewhere), so WDYT about moving a (protected) method?

```suggestion
        tracker = build_tracker
```

```ruby
  protected

  def build_tracker
    Thread.current[ProxyConfigAffectingChanges::TRACKER_NAME] = ProxyConfigAffectingChanges::Tracker.new
  end
```

Reply:
Since the other tests are using it as well I moved it to the static method instead